examples/fft.dx

```diff
@@ -0,0 +1,215 @@
+'# Fast Fourier Transform
+For arrays whose size is a power of 2, we use a radix-2 algorithm based
+on the [Fhutark demo](https://github.com/diku-dk/fft/blob/master/lib/github.com/diku-dk/fft/stockham-radix-2.fut#L30).
+For non-power-of-2 sized arrays, it uses
+[Bluestein's Algorithm](https://en.wikipedia.org/wiki/Chirp_Z-transform),
+which calls the power-of-2 FFT as a subroutine.
+
+
+'## General helper functions
+
+def isPowerOf2 (x:Int) : Bool =
+  -- A fast trick based on bitwise AND.
+  -- Note: The bitwise and operator (.&.)
+  -- is only defined for Byte, which is why
+  -- I use %and here.
+  -- A test below verifies that this works on
+  -- more than 8 bit integers.
+  -- TODO: Make (.&.) polymorphic.
+  if x == 0
+    then False
+    else 0 == %and x (x - 1)
+
+def intpow [Mul a] (base:a) (power:Int) : a =
+  -- Todo: Use Knuth's power trick,
+  -- which can give an asymptotic speedup.
+  reduce one (*) (for a:(Fin power). base)
+
+def intlog2 (x:Int) : Int =
+  yieldState (-1) \ansRef.
+    runState 1 \cmpRef.
+      while do
+        if x >= (get cmpRef)
+          then 
+            ansRef := (get ansRef) + 1
+            cmpRef := %shl (get cmpRef) 1
+            True
+          else
+            False
+
+def intpow2 (power:Int) : Int = %shl 1 power
+
+def nextpow2 (x:Int) : Int = case isPowerOf2 x of
+  True -> x
+  False -> intpow2 (1 + intlog2 x)
+
+def reflect (i:n) : n =
+  s = size n
+  (s - 1 - ordinal i)@n
```

Comment:
Ok.  I also got rid of all the rest of the `@`.  Surprisingly to me, it made compilation much faster.

Reply:
I haven't looked at the IR in quite a while, but when I did recently I realized that since we've made errors use IO for printing in the frontend, any invocation of `@` got _really complicated_. cc @dougalm